src/main/java/io/xpring/xrpl/model/XRPTransaction.java

```diff
@@ -0,0 +1,173 @@
+        return ImmutableXRPTransaction.builder();
+    }
+
+    /**
+     * @return The unique address of the account that initiated the transaction.
+     */
+    String account();
+
+    /**
+     * @return (Optional) Hash value identifying another transaction.
+     * If provided, this transaction is only valid if the sending account's previously-sent
+     * transaction matches the provided hash.
+     */
+    @Nullable byte[] accountTransactionID();
+
+    /**
+     * @return Integer amount of XRP, in drops, to be destroyed as a cost for distributing
+     * this transaction to the network.
+     */
+    Long fee();
+
+    /**
+     * @return (Optional) Set of bit-flags for this transaction.
+     */
+    @Nullable Integer flags();
+
+    /**
+     * @return (Optional; strongly recommended) Highest ledger index this transaction can appear in.
+     * Specifying this field places a strict upper limit on how long the transaction can wait
+     * to be validated or rejected.
+     */
+    @Nullable Integer lastLedgerSequence();
+
+    /**
+     * @return (Optional) Additional arbitrary information used to identify this transaction.
+     */
+    @Nullable List<XRPMemo> memos();
+
+    /**
+     * @return The sequence number of the account sending the transaction.
+     * A transaction is only valid if the Sequence number is exactly 1 greater
+     * than the previous transaction from the same account.
+     */
+    Integer sequence();
+
+    /**
+     * @return (Optional) Array of objects that represent a multi-signature which authorizes this transaction.
+     */
+    @Nullable List<XRPSigner> signers();
+
+    /**
+     * @return Hex representation of the public key that corresponds to the private key used to sign this transaction.
+     * If an empty string, indicates a multi-signature is present in the Signers field instead.
+     */
+    byte[] signingPublicKey();
+
+    /**
+     * @return (Optional) Arbitrary integer used to identify the reason for this payment,
+     * or a sender on whose behalf this transaction is made.
+     * Conventionally, a refund should specify the initial payment's SourceTag as the refund payment's DestinationTag.
+     */
+    @Nullable Integer sourceTag();
+
+    /**
+     * @return The signature that verifies this transaction as originating from the account it says it is from.
+     */
+    byte[] transactionSignature();
+
+    /**
+     * @return The type of transaction.
+     */
+    TransactionType type();
+
+    /**
+     * @return an XRPPayment object representing the additional fields present in a PAYMENT transaction.
+     * @see "https://xrpl.org/payment.html#payment-fields"
+     */
+    XRPPayment paymentFields();
+
+    static XRPTransaction from(Transaction transaction) {
+        String account = transaction.getAccount().getValue().getAddress();
+
+        ByteString accountTxnIDByteString = transaction.getAccountTransactionId().getValue();
```

Comment:
I'd shy away from using `Txn` because we use `Transaction` everywhere else and I don't think it's a really common abbreviation.